processor/src/chiplets/tests.rs

```diff
@@ -117,11 +117,11 @@ fn build_trace(
     process.execute_code_block(&program, &CodeBlockTable::default()).unwrap();
 
     let (trace, _, _) = ExecutionTrace::test_finalize_trace(process);
-    let trace_len = get_trace_len(&trace) - ExecutionTrace::NUM_RAND_ROWS;
+    let trace_len = trace.num_rows() - ExecutionTrace::NUM_RAND_ROWS;
 
     (
-        trace[CHIPLETS_RANGE]
-            .to_vec()
+        trace
+            .get_column_range(CHIPLETS_RANGE)
```

Comment:
Here we end up cloning some columns of `trace`, and then dropping `trace`, which isn't great. But the previous implementation did that, so we don't worsen the performance, and we can leave it for another PR to improve on it. And the compiler might even optimize that `clone()` away anyways.

Reply:
Yes - not critical to improve, especially since this is just for testing.